binary.mk

```diff
@@ -0,0 +1,33 @@
+# Makefile rules for compiling and linking binary file
+# supported external variables:
+# - NAME - component/target binary name
+# - SRCS - list of source files
+# - STATIC_LIBS - names of the libs to link the binary against
+
+### should be available in phoenix-rtos-build for usage in all projects
+
+# linking dla prerequisites
+OBJS := $(patsubst %.c,$(PREFIX_O)%.o,$(SRCS))
+RESOLVED_LIBS:= $(patsubst %,$(PREFIX_A)%.a, $(STATIC_LIBS))
+
+# compilation prerequisites - component dependency only
+$(OBJS) : | $(STATIC_LIBS)
+
+# dynamically generated dependencies (file-to-file dependencies)
+DEPS := $(patsubst %.c,$(PREFIX_O)%.c.d,$(SRCS))
+-include $(DEPS)
+
+# rule for linking binary
+# NOTE: if applied globally, we could remove the $(LINK) variable and put explicit commands here
+$(PREFIX_PROG)$(NAME): $(OBJS) $(RESOLVED_LIBS)
+	$(LINK)
+
+# create phony target for dependencies
+.PHONY: $(NAME)
+$(NAME): $(PREFIX_PROG_STRIPPED)$(NAME)
+
+
+# TODO: should we always add targets to all?
+# the other approach would be to include always all Makefiles and add them to "all" on per-TARGET basis
```

Comment:
What I like in all per TARGET basis is that this will looks much simpler (especially with PHONY targets).
But wonder if you would include all modules Makefiles explicitly or want to write wildcard rule (not sure if it's possilble and advisable)? 
Also – don't it increase Makefile processing time?

On the other hand maybe could work better with tab completion?